components/brave_shields/browser/tracking_protection_service.cc

```diff
@@ -90,6 +95,50 @@ bool TrackingProtectionService::ShouldStartRequest(const GURL& url,
   return false;
 }
 
+bool TrackingProtectionService::ShouldStoreState(HostContentSettingsMap* map, const GURL& top_origin_url, 
+  const GURL& origin_url) {
+
+  if (!first_party_storage_trackers_initailized_) {
+    LOG(INFO) << "First party storage trackers not initialized";
+    return true;
+  }  
+  std::string host = origin_url.host();
+
+  bool allow_brave_shields = IsAllowContentSetting(
+      map, top_origin_url, origin_url, CONTENT_SETTINGS_TYPE_PLUGINS,
+      brave_shields::kBraveShields);
+
+  bool allow_trackers = IsAllowContentSetting(
+      map, top_origin_url, origin_url, CONTENT_SETTINGS_TYPE_PLUGINS, 
+      brave_shields::kTrackers);
+
+  bool denyStorage = std::find(first_party_storage_trackers_.begin(), first_party_storage_trackers_.end(), host) 
+        != first_party_storage_trackers_.end();
+
+  return !(allow_brave_shields && !allow_trackers && denyStorage);
```

Comment:
can you elaborate the check of kBraveShields here?

Reply:
`STP` is bundled with `tracking-protection`.  If the `brave_shields` are disabled, `STP` checks should not block storage. This is under discussion, will update as soon as we reach consensus. Ideally, there should be a separate global switch to disable STP to debug web-compat issues.